### ISSUE TITLE
KeepAlive

### ISSUE THREAD

**Issue by alberk8** [dotnet/MQTTnet#1513]
### Verification

Before opening a bug make sure that the following conditions are met.

1. Performance issues are also appearing in RELEASE mode of your application.
2. An increased memory consumption or high CPU load also happens when your application runs in RELEASE mode AND logging is DISABLED.
3. Client issues with not received messages, connection drops etc. can be reproduced via another client application like "MQTTnetApp" (https://github.com/chkr1011/MQTTnetApp) or similar.
4. The bug also appears in the VERY LATEST version of this library. There is no support for older version (due to limited resources) but pull requests for older versions are welcome.

### Describe the bug
I have set the KeepAlivePeriod to 20 seconds but it does not seems to trigger a disconnect when the network is down.   Currently a disconnect will only trigger after 18 minutes of the network going down.

### Which component is your bug related to?

- ManagedClient version 4.1.0.247 on dotnet 6.0

### To Reproduce
1. Using this version of MQTTnet 4.1.0.247 on dotnet 6.0.
```
 var options = new MqttClientOptionsBuilder()
           .WithClientId(_clientID)
           .WithTcpServer((opt) =>
           {
               opt.Server = "broker.emqx.io";
               opt.Port = 8883;
               
               opt.TlsOptions = new MqttClientTlsOptions
               {
                   UseTls = true,
                   SslProtocol =
                       System.Security.Authentication.SslProtocols.Tls12 |
                       System.Security.Authentication.SslProtocols.Tls13,
               };

           })
           .WithKeepAlivePeriod(TimeSpan.FromSeconds(20))
           .WithTimeout(TimeSpan.FromSeconds(10))
           .Build();
```

### Expected behavior
The keepalive to kick in very 20 seconds and to trigger the disconnect event when there is no response.

### Additional context / logging
Include debugging or logging information here:

```batch
\\ Put your logging output here.
```
### Code example
Please provide full code examples below where possible to make it easier for the developers to check your issues.

**Ideally a Unit Test (which shows the error) is provided so that the behavior can be reproduced easily.**
 
```csharp
\\ Put your code here.
```

**Comment by ThomasSeidenbecher** [comment]
I can confirm this issue. Looks like the async-wait for the ping reply packet is performed without any timeout.

**Comment by alberk8** [comment]
There is a change on the [Ping](https://github.com/dotnet/MQTTnet/pull/1495) and merged  but not release on the latest version.

**Comment by chkr1011** [comment]
This is indeed already merged and will be fixed in version 4.1.1 which will be released soon.
